models/news.models.js

```diff
@@ -23,7 +23,24 @@ exports.selectTopics = () => {
     });
 };
 
-exports.selectArticles = () => {
+exports.selectArticles = (reqQuery) => {
+    const qualifiedQuery = ['topic'];
+  
+    if (Object.keys(reqQuery).length > 0){
+        const queryArr = Object.keys(reqQuery);
+        for (let i = 0; i < queryArr.length; i++) {
+            if (!qualifiedQuery.includes(queryArr[i])){
+                return Promise.reject({msg: 'Not Found'});
+            }
+        }
+    };
+
+    let topicQuery = '', bodyColumn = '';
+    if (reqQuery.topic){
```

Comment:
This wont protect you from SQL injection, you need a way of checking for valid topics (a utility function maybe) and check that the topic value is valid that way as currently `topic=;DROP DATABASE`; would still work just fine